controller/celebrant.controller.js

```diff
@@ -0,0 +1,253 @@
+        });
+    }
+};
+
+//get all celebrants
+const getAllCelebrants = async (req, res) => {
+    try {
+        // Fetch all celebrants that are not marked as deleted
+        const getCelebrantsQuery = {
+            text: 'SELECT * FROM celebration.celebrants WHERE is_deleted = false'
+        };
+
+        const { rows } = await pool.query(getCelebrantsQuery);
+
+        if (rows.length === 0) {
+            return res.status(404).json({ message: "No celebrants found" });
+        }
+
+        res.status(200).json({
+            msg: "Successful!",
+            data: rows
+        });
+    } catch (error) {
+        console.error("Error fetching celebrants", error);
+        return res.status(500).json({
+            message: "Internal server error"
+        });
+    }
+};
+
+//update a celebrant
+const updateCelebrant = async (req, res) => {
+    const id = req.params.id;
+    const {username,gender,phone_number,email,birthdate,channel,is_active} = req.body;
+
+    try {
+        // Fetch existing celebrant details
+        const getExistingCelebrantQuery = {
+            text: 'SELECT * FROM celebration.celebrants WHERE id = $1',
+            values: [id]
+        };
+
+        const { rows: existingCelebrant } = await pool.query(getExistingCelebrantQuery);
+
+        if (existingCelebrant.length === 0) {
+            return res.status(404).json({ message: "Celebrant not found" });
+        }
+
+        // Update celebrant excluding the username
+
+        const updateACelebrant ={text: 'UPDATE celebration.celebrants SET gender = $1, phone_number = $2, email = $3,birthdate = $4, channel = $5 ,is_active = $6 WHERE id =$7 RETURNING *' , 
+        values: [gender,phone_number,email,birthdate,channel, is_active, id]
+    };
+
+    const { rows } = await pool.query(updateACelebrant);
+
+     // Update the username if it's provided and different from the existing one
+     if (username && username !== existingCelebrant[0].username) {
+        const updateUsernameQuery = {
+            text: 'UPDATE celebration.celebrants SET username = $1 WHERE id = $2 RETURNING *',
+            values: [username, id]
+        };
+        const { rows: updatedUsernameCelebrant } = await pool.query(updateUsernameQuery);
+        rows[0].username = updatedUsernameCelebrant[0].username; // Update the username in the response
+    }
+
+    res.status(200).json({
+        msg: "Celebrant updated successfully!",
+        data: rows[0]
+    });
+    
+    }
+    catch (error) {
+        console.error("Error updating celebrant", error);
+        return res.status(500).json({
+          message: "internal server error",
+        });
+    }
+};
+
+
+ // delete a celebrant
+const deleteCelebrant = async (req, res) => {
+    const id = req.params.id;
+    try {
+          // Soft delete by setting the is_deleted flag to true
+        const deleteACelebrant = {text:'UPDATE celebration.celebrants SET is_deleted = true WHERE id = $1',
```

Comment:
**Soft delete** also requires resetting some of the fields (other than `is_deleted`) to some randomize values.